[News] 2021-07-13 (#636)
sider/sider-docs: docs/tools/cplusplus/cppcheck.md

    addon: "cert"
    bug-hunting: true
    parallel: true

>     suppressions-list: "suppressions.txt"

Reviewer: [trivial]

```suggestion
    suppressions-list: "your_suppressions.txt"
```

| [`enable`](#enable)                                                                           | `string`             | -                                |
| [`std`](#std)                                                                                 | `string`             | -                                |
| [`project`](#project)                                                                         | `string`             | -                                |
| [`language`](#language)                                                                       | `string`             | -                                |
| [`addon`](#addon)                                                                             | `string`, `string[]` | -                                |
| [`bug-hunting`](#bug-hunting)                                                                 | `boolean`            | `false`                          |
| [`parallel`](#parallel)                                                                       | `boolean`            | `false`                          |

> | [`suppressions-list`](#suppressions-list)                                                     | `string`             | `sider_recommended_cppcheck.txt` |

Reviewer: [suggestion] This default value is inaccurate. How about this?

```suggestion
| [`suppressions-list`](#suppressions-list)                                                     | `string`             | _(see below)_ |
```

Author: I also added this mark in the `rule` row of the PHPMD table.

- unusedFunction
- Whole program analysis (ctu=Cross Translation Unit)

### `suppressions-list`

This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:

```text
memleak
uninitvar
```

> You can read more about this file format in the [Cppcheck manual](https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file).

Author: > __[remark-lint]__ Link to https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file is dead ([view](https://sider.review/gh/repos/142577083/pulls/636#issue-5749131))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `no-dead-urls` | `warn` |
>
> You can [close this issue](https://sider.review/gh/repos/142577083/pulls/636#issue-5749131) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

The link is valid.

Reviewer: [suggestion] https://github.com/sider/sider-docs/pull/636/files#r668400505

How about adding a link to `sider_recommended_cppcheck.txt` in this section?

- unusedFunction
- Whole program analysis (ctu=Cross Translation Unit)

### `suppressions-list`

This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:

```text
memleak
uninitvar
```

> If omitted, Sider uses the [recommended ruleset](https://github.com/sider/runners/blob/HEAD/images/cppcheck/sider_recommended_cppcheck.txt).

Reviewer: [suggestion] How about avoiding duplication of the link to `sider_recommended_cppcheck.txt`?

```suggestion
If omitted, Sider uses the [recommended ruleset](#default-configuration-for-cppcheck).
```
